Source/Events.Store.MongoDB/DatabaseConnection.cs

```diff
@@ -19,6 +24,7 @@ public class DatabaseConnection
         /// <param name="configuration">A <see cref="IConfigurationFor{EventStoreConfiguration}"/> with database connection parameters.</param>
         public DatabaseConnection(IConfigurationFor<EventStoreConfiguration> configuration)
         {
+            RegisterCustomDiscriminators();
```

Comment:
Is this safe to do multiple times?

Reply:
It's not safe to do multiple times, that's why I added the `[SingletonPerTenant]` attribute to the class so that it only get's called once per tenant.
Not sure how it deals with multiple tenants, doesn't a single tenant own a single MongoDB database with it's own mappings and registered serializers etc which wouldn't clash then?

Reply:
Nope - they share the same code, so this would be called once for every tenant. If this thing is supposed to only be called once - I would suggest we use a static constructor (see https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/static-constructors)

Reply:
Added static constructor.